components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/correlation/CorrelationConfigManager.java

```diff
@@ -0,0 +1,194 @@
+/*
+ *
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  n compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.impl.correlation;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dto.CorrelationConfigDTO;
+import org.wso2.carbon.apimgt.impl.dto.CorrelationConfigPropertyDTO;
+import org.wso2.carbon.apimgt.impl.dto.EventHubConfigurationDto;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.logging.correlation.bean.CorrelationLogConfig;
+import org.wso2.carbon.logging.correlation.utils.CorrelationLogHolder;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+/**
+ * Correlation Config Manager to configure correlation components and
+ * invoke the internal API.
+ */
+public class CorrelationConfigManager {
+    public static final int RETRIEVAL_RETRIES = 15;
+    public static final int RETRIEVAL_TIMEOUT_IN_SECONDS = 15;
+    public static final String UTF8 = "UTF-8";
+    private static final String DENIED_THREADS = "deniedThreads";
+    private static final Log log = LogFactory.getLog(CorrelationConfigManager.class);
+    private static final CorrelationConfigManager correlationConfigManager = new CorrelationConfigManager();
+    private final EventHubConfigurationDto eventHubConfigurationDto;
+
+    public CorrelationConfigManager() {
```

Comment:
Since `CorrelationConfigManager` is a singleton, we need to have a private Constructor